### DISCUSSION TITLE
/the-basics/response.html

### DISCUSSION THREAD

**Discussion by giscus** [goravel/goravel#346] [Announcements]
# /the-basics/response.html

[[toc]] 简介 可以使用 ctx.Response() 在控制其中进行 HTTP 响应。 字符串 JSON 自定义返回 文件响应 下载文件 附加 Header 返回成功 自定义 Code 重定向 获取响应 可以获取响应的各种信息，一般用在 HTTP 中间件中： origin 包含一下方法： 方法名 作用 ----------- ---------...

https://www.goravel.dev/zh/the-basics/response.html

**Comment by luscio** [comment]
能不能像laravel那样优雅的处理异常

**Reply by hwbrzzl** [reply]
golang 中目前最优雅的方式就是老老实实 `return err` ~

**Comment by zerolethanh** [comment]
Do you support response stream?

**Reply by hwbrzzl** [reply]
cannot for now, but we can have it, created an issue for it: https://github.com/goravel/goravel/issues/449

**Reply by zerolethanh** [reply]
thanks you. It's very useful when we are in AI era. Can't wait 😉

**Comment by jxgame** [comment]
附加 Header 注意了。
前面要加一句 
ctx.Response().Header("Access-Control-Expose-Headers", "Your-Name")

这样才能在前端拿到。

**Reply by hwbrzzl** [reply]
是不是被 cors 过滤掉了

**Comment by klphp** [comment]
请教，如获取完用户数据后，怎么控制返回哪些字段。

**Reply by krishankumar01** [reply]
You could use the `Select` method in your ORM to get only the fields you need, or add a custom method in `User` model to return just those specific fields.

**Comment by wangxuancheng-dev** [comment]
ctx.Response().Json(http.StatusOK, struct {
  ID       uint `json:"id"`
  Name     string `json:"name"`
}{
  Id:      1,
  Front:   "Goravel",
})

这里要这样改吧
return ctx.Response().Json(http.StatusOK, struct {
		ID   uint   `json:"id"`
		Name string `json:"name"`
	}{
		ID:   1,
		Name: "Goravel",
	})

**Comment by hsemix** [comment]
How would I do a flash-message on a redirect that only displays once and the next time you load the uri, it's gone?

**Reply by krishankumar01** [reply]
Yes, you can use `session.Flash` for that.

Documentation: [https://www.goravel.dev/the-basics/session.html#flash-data](https://www.goravel.dev/the-basics/session.html#flash-data)

**Comment by etrokal** [comment]
How would I return an error response? Like a 500 or 404?

**Reply by hwbrzzl** [reply]
You can set `http.Status*`

**Reply by etrokal** [reply]
So I have to deal with the error response directly on the action handler? Can I return an error to deal with it inside an error middleware?

**Reply by hwbrzzl** [reply]
Yes, add the error code on the action handler.

**Comment by bgdar** [comment]
I don't know what's wrong when using redirect, instead of being directed to the page I want, I get a status code of 500 , example in the controller : `
	return ctx.Response().Redirect(http.StatusTemporaryRedirect, "/user/profile")`

**Reply by hwbrzzl** [reply]
Hey @bgdar Could you create a PR on goravel/example? I cannot reproduce it locally.

**Reply by bgdar** [reply]
Sorry sir, do I send my project code example to goravel/example as a pull request?

**Reply by hwbrzzl** [reply]
You just need to create a PR with the minimal code example.
